src/utils/data-stream.ts

```diff
@@ -0,0 +1,48 @@
+import { Encoder } from './encoder.js';
+import { Readable } from 'readable-stream';
```

Comment:
`import type` might be more clear here

Reply:
Can't import just type here, because we do use it beyond just a type declaration, e.g. the constructor call.